proto/p4/config/p4info.proto

```diff
@@ -55,6 +53,22 @@ message Preamble {
   repeated string annotations = 4;
 }
 
+// used to group all extern instances of the same type in one message
+message Extern {
+  // the extern_type_id is assigned during compilation. It is likely that this
+  // id will in fact come from a P4 annotation to the extern declaration and
+  // that each vendor will receive a prefix to avoid collisions.
+  uint32 extern_type_id = 1;
+  string extern_type_name = 2;
+  repeated ExternInstance instances = 3;
+}
+
+message ExternInstance {
+  Preamble preamble = 1;
+  // specific to the extern type, declared in a separate proto file
+  google.protobuf.Any info = 2;
```

Comment:
Would it be useful to add a comment clarifying that the separate proto file is likely to be vendor dependent?